common/lib/xmodule/xmodule/modulestore/search.py

```diff
@@ -112,18 +116,54 @@ def find_path_to_course():
                     section_desc = modulestore.get_item(path[path_index])
                     # this calls get_children rather than just children b/c old mongo includes private children
                     # in children but not in get_children
-                    child_locs = [c.location for c in section_desc.get_children()]
+                    child_locs = get_child_locations(section_desc, request, course_id)
                     # positions are 1-indexed, and should be strings to be consistent with
                     # url parsing.
-                    position_list.append(str(child_locs.index(path[path_index + 1]) + 1))
+                    if path[path_index + 1] in child_locs:
+                        position_list.append(str(child_locs.index(path[path_index + 1]) + 1))
             position = "_".join(position_list)
 
     return (course_id, chapter, section, vertical, position, path[-1])
 
 
+def get_child_locations(section_desc, request, course_id):
+    """
+    Returns all child locations for a section. If request is present and learner is accessing, we have to
```

Comment:
docstring can still be improved. Something like `get all the children locations, except the staff only child when accessed or masqueraded as learner`. That's just for an idea.